client/components/tinymce/plugins/wpcom-view/views/embed/index.js

```diff
@@ -45,7 +48,12 @@ export default class EmbedViewManager extends EventEmitter {
 		super.addListener( event, listener );
 		if ( 'change' === event && 1 === this.listeners( event ).length ) {
 			this.store = store;
-			this.unsubscribeStore = store.subscribe( this.sitesListener );
+			this.selectedSiteSubscriberId = uniqueId();
+			store.dispatch( {
+				type: SELECTED_SITE_SUBSCRIBE,
+				subscriberId: this.selectedSiteSubscriberId,
```

Comment:
Forcing the caller to generate a unique id here works, but I do wonder if we can move this into the middleware. Maybe if we force listeners to have reference quality when they dispatch unsubscribe? I don't think this is a blocker, but it would relieve some of the burden on folks dispatching this.

So the actions might look like:
```jsx
{ type: SELECTED_SITE_SUBSCRIBE, listener }
{ type: SELECTED_SITE_UNSUBSCRIBE, listener }
```

@mcsf what do you think?